assessments/projects/openfga/joint-assessment.md

```diff
@@ -0,0 +1,647 @@
+|Impact (High/ Med/ Low)|High|
+|Likelihood (High/ Med/ Low)|Low|
+
+|Summary|Helm chart runs containers with higher privilege by default.|
+|--|--|
+|Discovered in self-assessment?|No|
+|Weakness|Unauthorized access to openfga data. Ability to both view and manipulate data.|
+|MITRE classification|TA0002: Execution -> T1203: Exploitation for Client Execution<br>TA0003: Persistence -> T1098: Account Manipulation<br>TA0004: Privilege Escalation -> T1548: Abuse Elevation Control Mechanism<br>TA0004: Privilege Escalation -> T1134: Access Token Manipulation<br>TA0004: Privilege Escalation -> T1098: Account Manipulation<br>TA0004: Privilege Escalation -> T1611: Escape to Host<br>TA0005: Defense Evasion -> T1548: Abuse Elevation Control Mechanism<br>TA0005: Defense Evasion -> T1134: Access Token Manipulation|
+|Actors|openfga.server|
+|Suggested Mitigation|Could the defaults for all install scripts be set to run the openfga server with limited permissions?<br>In the case of helm chart, this would achieve: <ol><li>Not running server as root</li> <li>Not allowing privilege escalation</li> <li>Not allowing access to system calls unless required</li> <li>Setting filesystem to readonly </li> <li>Limiting access to mounted filesystems</li></ol><br>This would greatly reduce the attack surface area. |
+|Impact (High/ Med/ Low)|High|
+|Likelihood (High/ Med/ Low)|Low|
+
+|Summary|An external dependency to groundnuty/k8s-wait-for is pinned using tag.|
+|--|--|
+|Discovered in self-assessment?|No|
+|Weakness|Source tags can be overwritten in case of a supply chain attack and a compromised image may be pulled down. The risk is greater in the case of external, third party dependencies not under the projects control.|
+|MITRE classification|TA0001: Initial Access -> T1195: Supply Chain Compromise|
+|Actors|openfga.server|
+|Suggested Mitigation|Pin the dependency using SHA tag for the container image.|
+|Impact (High/ Med/ Low)|Low|
+|Likelihood (High/ Med/ Low)|Low|
+
+### Other findings
+
+This section has other findings that could not be classified 
+in earlier parts. It includes exploits such as server DDOS and 
+potential leakage of information about application landscape.
+
+|Summary|Playground link as well as shape of API identifies the openfga server.|
+|--|--|
+|Discovered in self-assessment?|No|
+|Weakness|An attacker performing reconnaissance will be able to identify if a service is the openfga server, a high-value target.|
+|MITRE classification|TA0043: Reconnaissance -> T1592: Gather Victim Host Information|
+|Actors|openfga.server|
+|Suggested Mitigation|Not sure if there is a remediation, given it’s the case for any API server that endpoints will return a 403 rather than 404 for unauthenticated access. An attacker iterating through the expected shape of the API will be able to identify an openfga server.|
+|Impact (High/ Med/ Low)|Low|
+|Likelihood (High/ Med/ Low)|Low|
+
+|Summary|Usage of PII as part of models.|
```

Comment:
```suggestion
|Summary|Usage of PII as part of tuples.|
```

I can't imagine a scenario where the _model_ contains PII.

Reply:
Very valid comment :-), corrected.